src/base58.rs

```diff
@@ -0,0 +1,182 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt::{Display, Formatter},
+    ops::{Deref, DerefMut},
+    str::FromStr,
+};
+
+#[serde_with::serde_as]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize, Hash, PartialOrd, Ord)]
+#[serde(transparent)]
+pub struct AsString<T>(#[serde_as(as = "serde_with::DisplayFromStr")] pub T)
+where
+    T: Display + FromStr,
+    <T as FromStr>::Err: Display;
+
+impl<T> Deref for AsString<T>
+where
+    T: Display + FromStr,
+    <T as FromStr>::Err: Display,
+{
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for AsString<T>
+where
+    T: Display + FromStr,
+    <T as FromStr>::Err: Display,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, PartialOrd, Ord)]
+pub struct Base58<T>(pub T);
```

Comment:
Intended for use instead of `Wrapped` and `Base58` wrapper type in places, where base58 representation in necessary.